src/java_bytecode/java_object_factory.cpp

```diff
@@ -139,6 +142,8 @@ class java_object_factoryt
     code_blockt &assignments,
     bool create_dynamic_objects,
     const pointer_typet &substitute_pointer_type);
+
+  std::shared_ptr<select_pointer_typet> pointer_type_selector;
```

Comment:
This new field is alone at the end of the class, while other fields are together at the beginning. Maybe this could be moved together with other fields (easier to read and find)? Would it be possible to document the field? What it is used for?